apps/mobile/src/store/settings/settings.ts

```diff
@@ -0,0 +1,128 @@
+import { useSelector } from 'react-redux';
+
+import { whenTheme } from '@/utils/when-theme';
+
+import {
+  AccountDisplayPreference,
+  BitcoinUnit,
+  DefaultNetworkConfigurations,
+  FiatCurrency,
+  WalletDefaultNetworkConfigurationIds,
+} from '@leather.io/models';
+
+import { useAppDispatch } from '../utils';
+import {
+  selectAccountDisplayPreference,
+  selectAnalyticsEnabled,
+  selectAppAuthEnabled,
+  selectBitcoinUnit,
+  selectConversionUnit,
+  selectNetwork,
+  selectTheme,
+  selectWalletSecurityLevel,
+} from './settings.read';
+import {
+  userChangedAccountDisplayPreference,
+  userChangedBitcoinUnit,
+  userChangedConversionUnit,
+  userChangedNetwork,
+  userChangedTheme,
+  userChangedWalletSecurityLevel,
+  userToggledAnalyticsEnabled,
+  userToggledAppAuthEnabled,
+} from './settings.write';
+
+export const defaultNetworks = ['mainnet', 'testnet', 'signet'] as const;
+
+export const defaultThemes = ['light', 'dark', 'system'] as const;
+export type ThemeStore = (typeof defaultThemes)[number];
+export type Theme = Exclude<ThemeStore, 'system'>;
+
+export type WalletSecurityLevel = 'undefined' | 'secure' | 'insecure';
+
+export interface SettingsState {
+  accountDisplayPreference: AccountDisplayPreference;
+  analyticsEnabled: boolean;
+  appAuthEnabled: boolean;
+  bitcoinUnit: BitcoinUnit;
+  conversionUnit: FiatCurrency;
+  createdOn: string;
+  network: DefaultNetworkConfigurations;
+  theme: ThemeStore;
+  walletSecurityLevel: WalletSecurityLevel;
+}
+
+export const initialState: SettingsState = {
+  accountDisplayPreference: 'ns',
+  analyticsEnabled: false,
+  appAuthEnabled: false,
+  bitcoinUnit: 'bitcoin',
+  conversionUnit: 'USD',
+  createdOn: new Date().toISOString(),
+  network: WalletDefaultNetworkConfigurationIds.mainnet,
+  theme: 'system',
+  walletSecurityLevel: 'undefined',
+};
+
+export function useSettings() {
+  const dispatch = useAppDispatch();
+  const network = useSelector(selectNetwork);
+  const walletSecurityLevel = useSelector(selectWalletSecurityLevel);
+  const systemTheme = useColorScheme();
+  const accountDisplayPreference = useSelector(selectAccountDisplayPreference);
+  const analyticsEnabled = useSelector(selectAnalyticsEnabled);
+  const appAuthEnabled = useSelector(selectAppAuthEnabled);
+  const bitcoinUnit = useSelector(selectBitcoinUnit);
+  const conversionUnit = useSelector(selectConversionUnit);
+  const themeStore = useSelector(selectTheme);
+  const theme = (themeStore === 'system' ? systemTheme : themeStore) ?? 'light';
+
+  return {
+    accountDisplayPreference,
+    analyticsEnabled,
+    appAuthEnabled,
+    bitcoinUnit,
+    conversionUnit,
+    network,
+    theme,
+    themeStore,
+    walletSecurityLevel,
+    whenTheme: whenTheme(theme),
+    changeAccountDisplayPreference(type: AccountDisplayPreference) {
+      dispatch(userChangedAccountDisplayPreference(type));
+    },
+    toggleAnalyticsEnabled() {
+      dispatch(userToggledAnalyticsEnabled());
+    },
```

Comment:
Not big on this name, kinda sounds like we're enabling it, but also toggling it which doesn't make sense. We call the store item `analyicsEnabled` but really what's happening here is the user is changing the setting. So this is almost exposing a store detail—only the name, but still something the rest of the app doesn't care about. `toggleAnalyticsPreference`?

Reply:
Agree, was struggling with it and tried a few things. Will change.